### PR TITLE
Disable interrupts when reading RX fifo (RX lockup)

### PR DESCRIPTION
**Pull request by CapnBry**
This pull is more work to eliminate #553. It forces interrupts off while the Radio driver is reading the RF data during a receive interrupt on all MCU and RF platforms.

### Root cause analysis
Running an R9M TX + R9Mini RX, I was able to reproduce the lockup at a rate of about once every other day. 100% of the time the code was locked in the RXnbISR reading the FIFO. The number of bytes into the receive varied, but always it was hanging busywaiting for the SPI peripheral to set the `SPI_SR_RXNE` flag (RX not empty). There's no getting out of this loop due to the timeout existing _outside_ this busywait; not that anything checks the SPI transfer result anyway.
stack trace

What is happening is that the hwTimer Tock is firing **while we are in the middle of a packet read**. I have verified this by setting a flag when in this critical code and checking it in the Tock handler and incrementing a counter. This happens every minute or two, sometimes in bursts of 2-3, but _usually_ does not lock up because there's more code there than just the SPI transfer and we only run into a problem 1) on FHSS hop 2) on telemetry send 3) if changing PPM offset. All of these both write data to SPI (which will likely be ineffective due to being mid-transfer) and then de-assert `nCS` to the RF chip which will prevent it from sending more data from the FIFO. Here is the simple "danger" counter code:
dangercode

Note that this isn't necessarily a packet of data from ExpressLRS as the interrupt sometimes can fire from other people's packets. You down with OPP? This interrupt is. The CRC will fail later, but if we can't get the 8 bytes from the SX12xx without locking up, we'll never know.

### Fix
I think disabling interrupts during this critical code is the only way to guarantee that the timer won't preempt the read. I considered only wrapping the FIFO read, but really that could mean any of the other SPI transfers would fail. Those would be ephemeral and just give us some bad RSSI/SNR or maybe not clear the RX interrupt but I think we should guard against that as well.

The other option would be to set the STM32 ext interrupt priority to be the same as the timer's, but I do not think that is necessary. The other code in the packet callback doesn't do any radio stuff or muck about with any global variables that would cause an issue.  If we are interested in the lowest possible latency for our channels data, then it would make sense to make their priority equal, because the way I have it now the Tock timer ISR will fire immediately between the FIFO read and the packet actually being sent in `ProcessRFPacket()`. I didn't make the design decision to make them different priorities so that decision is above my pay grade and I am not entirely sure it is necessary.

There could be other solutions? Let me know in the comments and like and subscribe and hit that bell.

### External Factors

Having an RF-noisy environment was key to reproducing this lockup. I was getting one lockup every two days, but then I added a second Team900 TX to the mix, on a different bind phrase. After adding the second TX I got 3 lockups in 4 hours, of of which were hung in the RX ISR. The extra TX generates more RX ISR when the receivers are on the same channel I believe, and that increases the chances of being in the critical code when the timer fires.

### Extras
* IRQneedsClear has been removed due to every case of calling `ClearIRQFlags()` being immediately preceded by setting it to true. Less code means less time with interrupts disabled.
* `GetLastPacketSNR()` did a float divide instead of integer divide and then converted it to an int. Again, less code means less time with interrupts disabled.
* The R9MM variant had two places it set interrupt priority (that were identical) so I've removed one of them. All other variants were checked and they only have one instance of these lines.

### Thanks
Thanks to discord pro JamesK for their feedback and suggestions about the ISR collisions and leading me to quantify just how frequently they occur. I was fairly sure Tock would preempt the RX ISR very rarely but JamesK was all like "It is probably happening a lot more frequently than you think" and that turns out to be right. Great to have you with us, JamesK!

### CODE REVIEW

**Review by cruwaller — APPROVED** [review]
LGTM

Thanks a lot for the effort you have put to debug this!

**Comment by JBKingdon** [comment]
Great investigation. Thanks for the shout out, made my day :)

File: src/variants/R9MM/variant.h [hunk 1]
```diff
@@ -122,9 +122,6 @@ extern "C" {
 /*----------------------------------------------------------------------------
  *        Arduino objects - C++ only
  *----------------------------------------------------------------------------*/
- 
- #define TIM_IRQ_PRIO       3
- #define EXTI_IRQ_PRIO      4
```
Comment (CapnBry) [inline src/variants/R9MM/variant.h:125]: ^^ Note that just 9 lines above this are the exact same two lines

Comment (JBKingdon) [reply src/variants/R9MM/variant.h:125]: And those lines were new for V1

Comment (JBKingdon) [reply src/variants/R9MM/variant.h:125]: Unfortunately they came in as part of the humongous #505 PR, so it's hard to find out what the rationale was.

**Comment by AlessandroAU** [comment]
Thanks Cap for all the work and the amazing writeup!

**Pull request merged**